src/main/java/core/basesyntax/service/Figure.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service;
+
+import core.basesyntax.supplier.ColorSupplier;
+import core.basesyntax.supplier.FigureSupplier;
+
+public abstract class Figure implements FigureBehaviourService {
+    protected ColorSupplier colorSupplier = new ColorSupplier();
+    private String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    @Override
+    public void drawFigure() {
+        FigureSupplier figureSupplier = new FigureSupplier(new ColorSupplier());
```

Comment:
no need to instantiate FigureSupplier each time method was called,
lets extract it to the class-level field like `ColorSupplier`.

lets provide to FigureSupplier `ColorSupplier` that we already instantiated in class-level field.